winpr/libwinpr/sspi/Kerberos/kerberos.c

```diff
@@ -946,20 +963,29 @@ static SECURITY_STATUS SEC_ENTRY kerberos_InitializeSecurityContextA(
 	/* On first call allocate a new context */
 	if (new_context.ctx)
 	{
+		const KRB_CONTEXT empty = { 0 };
+
 		context = kerberos_ContextNew();
 		if (!context)
 		{
 			status = SEC_E_INSUFFICIENT_MEMORY;
 			goto cleanup;
 		}
-		CopyMemory(context, &new_context, sizeof(KRB_CONTEXT));
-	}
+		*context = new_context;
+		new_context = empty;
 
-	sspi_SecureHandleSetLowerPointer(phNewContext, context);
-	sspi_SecureHandleSetUpperPointer(phNewContext, KERBEROS_SSP_NAME);
+		sspi_SecureHandleSetLowerPointer(phNewContext, context);
+		sspi_SecureHandleSetUpperPointer(phNewContext, KERBEROS_SSP_NAME);
+	}
 
 cleanup:
 
+{
+	/* second_ticket is not allocated */
```

Comment:
Maybe my brain's still asleep, but I don't see the point of setting memory about to be de-allocated to zero

Reply:
Because the `krb5_cred` structure in this case is not used as a "credential"; rather it's just used to organize different pieces of information (the client and server principals for the new cred)